src/main/java/org/opensearch/ad/feature/AbstractRetriever.java

```diff
@@ -30,8 +32,49 @@
 import org.opensearch.search.builder.SearchSourceBuilder;
 
 public abstract class AbstractRetriever {
-    protected double parseAggregation(Aggregation aggregation) {
+    protected double parseAggregation(Aggregation aggregationToParse) {
         Double result = null;
+        /* example InternalSingleBucketAggregation: filter aggregation like
+           "t_shirts": {
+                "filter": {
+                    "bool": {
+                        "should": [
+                            {
+                                "term": {
+                                    "issueType": "foo"
+                                }
+                            }
+                            ...
+                            ],
+                            "minimum_should_match": "1",
+                            "boost": 1
+                    }
+                },
+                "aggs": {
+                    "impactUniqueAccounts": {
+                        "aggregation": {
+                            "field": "account"
+                        }
+                    }
+                }
+            }
+        
+            would produce an InternalFilter (a subtype of InternalSingleBucketAggregation) with a sub-aggregation
+            InternalCardinality that is also a SingleValue
+        */
+
+        if (aggregationToParse instanceof InternalSingleBucketAggregation) {
+            InternalAggregations bucket = ((InternalSingleBucketAggregation) aggregationToParse).getAggregations();
+            if (bucket != null) {
+                List<Aggregation> aggrs = bucket.asList();
+                if (aggrs.size() == 1) {
```

Comment:
Is it possible the `aggrs.size` > 1?

Reply:
Yes when people write multiple aggregations. But we only accept size 1. If greater than 1, we will throw EndRunException as before as one feature can have only one value at a time.